gnucash/import-export/import-main-matcher.c

```diff
@@ -845,6 +845,113 @@ gnc_gen_trans_assign_transfer_account_to_selection_cb (GtkMenuItem *menuitem,
     LEAVE("");
 }
 
+typedef enum
+{
+    DESCRIPTION = 0,
+    MEMO        = 1,
+    NOTES       = 2,
+} edit_field;
+
+static void
+gnc_gen_trans_edit_fields (GtkMenuItem *menuitem, GNCImportMainMatcher *info, edit_field field)
+{
+    GtkTreeView *treeview;
+    GtkTreeSelection *selection;
+    GtkTreeModel *model;
+    GtkTreeIter iter;
+    GNCImportTransInfo *trans_info;
+    GList *selected_rows, *l;
+    gboolean first, is_selection;
+    GList *refs = NULL;
+    gchar* messages[] = {_("Enter new Description"),_("Enter new Memo"),_("Enter new Notes")};
+    gchar* message = messages[field];
+    
+    ENTER("assign_transfer_account_to_selection_cb");
+    treeview = GTK_TREE_VIEW(info->view);
+    model = gtk_tree_view_get_model (treeview);
+    selection = gtk_tree_view_get_selection (treeview);
+    selected_rows = gtk_tree_selection_get_selected_rows (selection, &model);
+    first = TRUE;
+    is_selection = TRUE;
+    
+    DEBUG("Rows in selection = %i",
+          gtk_tree_selection_count_selected_rows (selection));
+    DEBUG("Entering loop over selection");
+    
+    if (gtk_tree_selection_count_selected_rows (selection) > 0)
+    {
+        gchar *new_field;
+        for (l = selected_rows; l != NULL; l = l->next)
+        {
+            gchar *path_str = gtk_tree_path_to_string (l->data);
+            GtkTreeRowReference *ref = gtk_tree_row_reference_new (model, l->data);
+            g_free (path_str);
+            refs = g_list_prepend (refs, ref);
+            if (gtk_tree_model_get_iter (model, &iter, l->data))
+            {
+                Split* first_split;
+                Transaction* trans;
+                gtk_tree_model_get (model, &iter, DOWNLOADED_COL_DATA, &trans_info, -1);
+                trans = gnc_import_TransInfo_get_trans (trans_info);
+                if (field == DESCRIPTION)
+                {
+                    new_field = gnc_input_dialog(info->main_widget, "", message, xaccTransGetDescription (trans));
+                    if (!new_field) break;
```

Comment:
Well I think we want to break. The user has selected the right transactions to edit, if they change their mind and hit cancel, they probably don't want to continue through the rest of the transactions.
Because of that, using switch(field) is cumbersome: not easy to break out of the loop in addition to breaking out of the switch().